engine/runtime/src/main/scala/org/enso/compiler/context/SuggestionBuilder.scala

```diff
@@ -375,57 +337,38 @@ final class SuggestionBuilder[A: IndexedSource](val source: A) {
     * @return the list of type arguments
     */
   private def buildTypeSignature(
-    bindings: Option[BindingAnalysis.Metadata],
     typeExpr: IR.Expression
   ): Vector[TypeArg] = {
-    def go(typeExpr: IR.Expression, args: Vector[TypeArg]): Vector[TypeArg] =
-      typeExpr match {
-        case IR.Application.Operator.Binary(left, op, right, _, _, _) =>
-          val arg = for {
-            leftArg  <- go(left.value, Vector()).headOption
-            rightArg <- go(right.value, Vector()).headOption
-          } yield TypeArg.Binary(leftArg, rightArg, op.name)
-          args :++ arg
-        case IR.Function.Lambda(List(targ), body, _, _, _, _) =>
-          val typeName = targ.name.name
-          val tdef = resolveTypeName(bindings, typeName)
-            .getOrElse(TypeArg.Value(QualifiedName.simpleName(typeName)))
-          go(body, args :+ tdef)
-        case IR.Application.Prefix(tfun, targs, _, _, _, _) =>
-          val appFunction = go(tfun, Vector()).head
-          val appArgs     = targs.flatMap(arg => go(arg.value, Vector()))
-          args :+ TypeArg.Application(appFunction, appArgs.toVector)
-        case tname: IR.Name =>
-          val typeName = tname.name
-          val tdef = resolveTypeName(bindings, typeName)
-            .getOrElse(TypeArg.Value(QualifiedName.simpleName(typeName)))
-          args :+ tdef
-        case _ =>
-          args
-      }
+    def go(expr: IR.Expression): TypeArg = expr match {
```

Comment:
`SuggestionBuilder` no longer needs `BindingAnalysis` and the mapping from `IR.Expression` to `TypeArg` seems like straightforward 1:1 conversion. Great simplification!